### PR TITLE
clang 6.0: zero initialization warnings in fei

### PR DESCRIPTION
**Pull request by IlyaSergeyev**
zero initialization, that not causes clang 6.0 warnings

### CODE REVIEW

File: _studio/mfx_lib/fei/h264_la/mfx_h264_la.cpp [hunk 1]
```diff
@@ -614,7 +614,7 @@ mfxStatus VideoENC_LA::InsertTaskWithReordenig(sLAInputTask &newTask, bool bEndO
     { 
         m_numLastB = 0;
         m_miniGop.resize(0);
-        sLAInputTask finished_task = {0};
+        sLAInputTask finished_task =sLAInputTask();
```
Comment (mgonchar) [inline _studio/mfx_lib/fei/h264_la/mfx_h264_la.cpp]: Here you use default initialization of POD struct to create temporary zero-initialized sLAInputTask object and then copy it. You don't need temporary object, just initialize the finished_task object in place:
```
sLAInputTask finished_task();
```

File: _studio/mfx_lib/fei/h264_la/mfx_h264_la.cpp [hunk 2]
```diff
@@ -750,14 +750,14 @@ mfxFrameSurface1 * VideoENC_LA::GetFrameToVME()
 
 static mfxStatus SubmitFrameLARoutine(void *pState, void * pParam, mfxU32 threadNumber, mfxU32 callNumber)
 {
+    (void)threadNumber;
```
Comment (mgonchar) [inline _studio/mfx_lib/fei/h264_la/mfx_h264_la.cpp]: nameless arguments better:
```
static mfxStatus SubmitFrameLARoutine(void *pState, void * pParam, mfxU32 /* threadNumber */, mfxU32 /* callNumber */)
```

File: _studio/mfx_lib/fei/h264_la/mfx_h264_la.cpp [hunk 3]
```diff
@@ -614,7 +614,7 @@ mfxStatus VideoENC_LA::InsertTaskWithReordenig(sLAInputTask &newTask, bool bEndO
     { 
         m_numLastB = 0;
         m_miniGop.resize(0);
-        sLAInputTask finished_task = {0};
+        sLAInputTask finished_task = {};
         m_syncTasks.push_back(finished_task);
```
Comment (dvrogozh) [inline _studio/mfx_lib/fei/h264_la/mfx_h264_la.cpp]: why not emplace_back()?

File: _studio/mfx_lib/fei/h264_la/mfx_h264_la.cpp [hunk 4]
```diff
@@ -725,7 +725,7 @@ mfxStatus VideoENC_LA::InsertTaskWithReordenigBPyr(sLAInputTask &newTask, bool b
     { 
         m_numLastB = 0;
         m_miniGop.resize(0);
-        sLAInputTask finished_task = {0};
+        sLAInputTask finished_task = {};
```
Comment (dvrogozh) [inline _studio/mfx_lib/fei/h264_la/mfx_h264_la.cpp]: same

File: _studio/mfx_lib/fei/h264_la/mfx_h264_la.cpp [hunk 5]
```diff
@@ -1160,7 +1151,7 @@ mfxStatus CopyRawSurfaceToVideoMemory(  VideoCORE &  core,
 {
     mfxExtOpaqueSurfaceAlloc const * extOpaq = GetExtBuffer(video);
 
-    mfxFrameData d3dSurf = {0};
+    mfxFrameData d3dSurf = {};
```
Comment (dvrogozh) [inline _studio/mfx_lib/fei/h264_la/mfx_h264_la.cpp]: `mfxFrameData d3dSurf{};` will probably be better

File: _studio/mfx_lib/fei/h264_la/mfx_h264_la.cpp [hunk 6]
```diff
@@ -1247,8 +1238,8 @@ mfxStatus VideoENC_LA::FreeUnusedVMEData(sVMEFrameInfo *pVME)
 }
 mfxStatus VideoENC_LA::SubmitFrameLA(mfxFrameSurface1 *pInSurface)
 {
-    mfxFrameSurface1*       inputSurface = 0;
-    sLAInputTask            currTask = { 0 };
+    mfxFrameSurface1*       inputSurface = nullptr;
+    sLAInputTask            currTask = {};
```
Comment (dvrogozh) [inline _studio/mfx_lib/fei/h264_la/mfx_h264_la.cpp]: `sLAInputTask            currTask{};` probably be better

**Comment by dvrogozh** [comment]
@mgonchar : please, review the updated change.

File: _studio/mfx_lib/fei/h264_la/mfx_h264_la.cpp [hunk 7]
```diff
@@ -614,8 +614,7 @@ mfxStatus VideoENC_LA::InsertTaskWithReordenig(sLAInputTask &newTask, bool bEndO
     { 
         m_numLastB = 0;
         m_miniGop.resize(0);
-        sLAInputTask finished_task = {0};
-        m_syncTasks.push_back(finished_task);    
+        m_syncTasks.emplace_back();
```
Comment (mgonchar) [inline _studio/mfx_lib/fei/h264_la/mfx_h264_la.cpp]: sLAInputTask doesn't have user defined constructor, so it won't be initialized, but code before assumed zero-initialized object. Pass empty initializer list as constructor argument, it should make the trick:
```
m_syncTasks.emplace_back(static_cast<std::initializer_list<sLAInputTask> >({})); // maybe you'll need slightly other cast, but I think this one should work
```

File: _studio/mfx_lib/fei/h264_la/mfx_h264_la.cpp [hunk 8]
```diff
@@ -725,8 +724,7 @@ mfxStatus VideoENC_LA::InsertTaskWithReordenigBPyr(sLAInputTask &newTask, bool b
...
```
Comment (mgonchar) [inline _studio/mfx_lib/fei/h264_la/mfx_h264_la.cpp]: same

**Comment by IlyaSergeyev** [comment]
I think, 
        sLAInputTask finished_task = {0};
        m_syncTasks.push_back(finished_task);   
is more readable and have zero initializer.

**Pull request merged**